packages/netlify-cms-backend-github/src/API.js

```diff
@@ -641,12 +646,18 @@ export default class API {
       const metadataPromise = this.retrieveMetadata(contentKey);
       const [commit, metadata] = await Promise.all([commitPromise, metadataPromise]);
       const { title, description } = options.parsedData || {};
+
+      // remove any existing media files
       const metadataFiles = get(metadata.objects, 'files', []);
-      const files = [...metadataFiles, ...filesList];
+      await Promise.all(
+        metadataFiles.map(file =>
+          this.deleteFile(file.path, options.commitMessage, { branch: branchName }),
```

Comment:
Instead of the re-uploading changes and deleting previously-stored media files, which will still occur if the draft change didn't add any new media file. Why don't we make use of the `uploaded` flag that is set [here](https://github.com/netlify/netlify-cms/blob/master/packages/netlify-cms-backend-github/src/API.js#L1048)  and then filter the loaded draft media files using this flag to get newly added media file(s), and then push the new media file(s) on draft update and concat here with previous ones.

Reply:
Thanks @barthc I haven't finished the part of deleting media files from the pr. I'll look into the `uploaded` flag.